### PR TITLE
SAAS-4494: Added Image Recipe storage volumes encrypted Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Image Recipe Storage Volumes Encrypted',
+    category: 'Imagebuilder',
+    domain: 'compute',
```
Comment (sadeed12345) [inline plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js:7]:
```suggestion
    domain: 'compute',
    severity: 'LOW'
```

File: plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                if (result) {
+                    helpers.addResult(results, 0,
+                        'Image recipe has ebs volume storage encrypted',
+                        region, resource);
+                } else {
+                    helpers.addResult(results, 2,
+                        'Image recipe does not have ebs volume storage encrypted',
+                        region, resource);
+                }
+
```
Comment (sadeed12345) [inline plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js:62]:
```suggestion
```

File: plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+module.exports = {
+    title: 'Image Recipe Storage Volumes Encrypted',
+    category: 'Image Builder',
+    domain: 'Compute',
+    severity: 'LOW',
+    description: 'Ensure that Image Recipe storage ebs volumes are encrypted.',
+    more_info: 'EC2 Image Builder is a fully managed AWS service that makes it easier to automate the creation, management, and deployment of customized, secure, and up-to-date server images that are pre-installed and pre-configured with software and settings to meet specific IT standards.',
+    link: 'https://docs.aws.amazon.com/imagebuilder/latest/userguide/data-protection.html',
+    recommended_action: 'Ensure that storage volumes for ebs are encrypted using AWS keys or customer managed keys in Image recipe',
+    apis: ['Imagebuilder:listImageRecipes', 'Imagebuilder:getImageRecipe'],
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js:13]: Add setting for aws-amanged or cmk level

File: plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Image Recipe Storage Volumes Encrypted',
+    category: 'Image Builder',
+    domain: 'Compute',
+    severity: 'LOW',
+    description: 'Ensure that Image Recipe storage ebs volumes are encrypted.',
+    more_info: 'EC2 Image Builder is a fully managed AWS service that makes it easier to automate the creation, management, and deployment of customized, secure, and up-to-date server images that are pre-installed and pre-configured with software and settings to meet specific IT standards.',
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js:10]: Let's make this description more specific to encryption

File: exports.js [hunk 5]
```diff
@@ -259,6 +259,8 @@ module.exports = {
         'bookmarkEncryptionEnabled'     : require(__dirname + '/plugins/aws/glue/bookmarkEncryptionEnabled.js'),
         'dataCatalogEncryptionEnabled'  : require(__dirname + '/plugins/aws/glue/dataCatalogEncryptionEnabled.js'),
 
+        'imgRcpStrgVlmEncrypted'        : require(__dirname + '/plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js'),
```
Comment (AkhtarAmir) [inline exports.js:262]:
```suggestion
        'imageRecipeVolumeEncrypted'    : require(__dirname + '/plugins/aws/imagebuilder/imgRcpStrgVlmEncrypted.js'),
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/imagebuilder/imageRecipeVolumeEncrypted.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Image Recipe Storage Volumes Encrypted',
+    category: 'Image Builder',
+    domain: 'Compute',
+    severity: 'LOW',
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/imageRecipeVolumeEncrypted.js:8]:
```suggestion
    severity: 'MEDIUM',
```

File: plugins/aws/imagebuilder/imageRecipeVolumeEncrypted.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    link: 'https://docs.aws.amazon.com/imagebuilder/latest/userguide/data-protection.html',
+    recommended_action: 'Ensure that storage volumes for ebs are encrypted using AWS keys or customer managed keys in Image recipe',
+    apis: ['Imagebuilder:listImageRecipes', 'Imagebuilder:getImageRecipe', 'KMS:listKeys', 'KMS:describeKey',
+        'KMS:listAliases'],
+    settings: {
+        image_recipe_ebs_volumes_desired_encryption_level: {
+            name: 'Image Recipe EBS Volumes Target Encryption Level',
+            description: 'In order (lowest to highest) awskms=AWS-managed KMS; awscmk=Customer managed KMS; externalcmk=Customer managed externally sourced KMS; cloudhsm=Customer managed CloudHSM sourced KMS',
+            regex: '^(awskms|awscmk|externalcmk|cloudhsm)$',
+            default: ''
```
Comment (AkhtarAmir) [inline plugins/aws/imagebuilder/imageRecipeVolumeEncrypted.js:20]:
```suggestion
            default: 'awscmk'
```

**Review by AkhtarAmir — APPROVED** [review]

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**